plugins/inputs/nginx_plus/nginx_plus.go

```diff
@@ -0,0 +1,607 @@
+package nginx_plus
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/errchan"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type NginxPlus struct {
+	Urls []string
+}
+
+var sampleConfig = `
+  ## An array of ngx_http_status_module or status URI to gather stats.
+  urls = ["http://localhost/status"]
+`
+
+func (n *NginxPlus) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NginxPlus) Description() string {
+	return "Read Nginx Plus' full status information (ngx_http_status_module)"
+}
+
+func (n *NginxPlus) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+	errChan := errchan.New(len(n.Urls))
+
+	for _, u := range n.Urls {
+		addr, err := url.Parse(u)
+		if err != nil {
+			return fmt.Errorf("Unable to parse address '%s': %s", u, err)
+		}
+
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			errChan.C <- n.gatherUrl(addr, acc)
+		}(addr)
+	}
+
+	wg.Wait()
+	return errChan.Error()
+}
+
+var tr = &http.Transport{
+	ResponseHeaderTimeout: time.Duration(3 * time.Second),
+}
+
+var client = &http.Client{
+	Transport: tr,
+	Timeout:   time.Duration(4 * time.Second),
+}
+
+func (n *NginxPlus) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
+	resp, err := client.Get(addr.String())
+	if err != nil {
+		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s returned HTTP status %s", addr.String(), resp.Status)
+	}
+	contentType := strings.Split(resp.Header.Get("Content-Type"), ";")[0]
+	switch contentType {
+	case "text/plain":
```

Comment:
This is a holdover from the previous PR - this should probably just be removed as the nginx plus status module will just return JSON.

Reply:
Yeah, just try to parse and don't worry about this being set correctly.